wordpress-mobile/WordPress-Android
# Issue/13326 fix scan backup menu animation

Parent issue #13326

This PR fixes two issues
1. List item added animation is played whenever you open MySite screen. This is solved by synchronously loading cached value (even when it's expired) of JetpackCapabilities. So unless the permissions changes or the user switches between sites the animation should not be played.
<img src="https://user-images.githubusercontent.com/2261188/107033236-4b008980-67b5-11eb-8c53-4c7955b7cc7c.gif" width="150px" />

2. The app would crash if fetchJetpackCapabilities was requested before the previous request finished. This is fixed in https://github.com/wordpress-mobile/WordPress-Android/commit/f74b76ecbf773d9b291270fd97c6310d8d22497a. New approach adds support for parallel request for different sites. If a duplicate request is initiated for the same site, the previous request is force finished with an error.

To test:
- clear app data
1. Log in with an account with multiple jetpack sites (eg. the one from the doc)
2. Select "pressable-jetpack-complete" site and notice "Scan" and "Backup" items are shown (with animation)
3. Open Reader and go back to MySite
4. Notice "Scan" and "Backup" items are shown (withOUT animation)
5. Select pressable-jetpack-daily-scan site and notice "Scan" is shown (backup disappears with an animation)
6. Kill the app
7. Re-open the app and notice "Scan" is shown (no animation is played)
----------------------------------
- clear app data
1. Log in with an account with multiple jetpack sites (eg. the one from the doc)
2. Enable MySiteImprovements feature flag and repeat steps 2-7 from the above section
-------------------------
1. Select pressable-jetpack-daily-backup
2. Open Activity Log and notice filters are shown
3. Select pressable-jetpack-daily-scan
4. Open Activity Log and notice filters are NOT shown

PR submission checklist:
